code/game/atoms_movable.dm

```diff
@@ -296,3 +299,69 @@
 		return null
 	return text2num(pickweight(candidates))
 
+
+/atom/movable/proc/set_glide_size(glide_size_override = 0, var/min = 0.9, var/max = WORLD_ICON_SIZE/2)
+	if (!glide_size_override || glide_size_override > max)
+		glide_size = 0
+	else
+		glide_size = max(min, glide_size_override)
+	
+	for (var/atom/movable/AM in contents)
+		AM.set_glide_size(glide_size, min, max)
+
+
+//This proc should never be overridden elsewhere at /atom/movable to keep directions sane.
+// Spoiler alert: it is, in moved.dm
+/atom/movable/Move(NewLoc, Dir = 0, step_x = 0, step_y = 0, var/glide_size_override = 0)
+	if (glide_size_override > 0)
+		set_glide_size(glide_size_override)
+	
+	if (Dir & (Dir - 1))
```

Comment:
Please comment in code what this whole condition body is doing.

Reply:
I didn't write this, I simply moved it from `mob_movement.dm` to here because it had no reason to be in `mob_movement.dm`.

I can still comment it if you want though.